src/realm/object-store/impl/realm_coordinator.cpp

```diff
@@ -309,6 +309,19 @@ ThreadSafeReference RealmCoordinator::get_unbound_realm()
 
 void RealmCoordinator::do_get_realm(Realm::Config config, std::shared_ptr<Realm>& realm,
                                     util::Optional<VersionID> version, util::CheckedUniqueLock& realm_lock)
+{
+    try {
```

Comment:
@tgoyne, you disagreed previously on this approach. In this PR I am actually only trying to catch possible errors while opening a realm, and delete all the files if something bad happened.